clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaClusterBasedRecordThrottler.java

```diff
@@ -23,30 +26,31 @@ public class KafkaClusterBasedRecordThrottler {
   // Kafka URL to records throttler
   private final Map<String, EventThrottler> kafkaUrlToRecordsThrottler;
   // Kafka URL to throttled records
-  protected Map<String, PubSubMessages<KafkaKey, KafkaMessageEnvelope, Long>> kafkaUrlToThrottledRecords;
+  protected Map<String, Map<PubSubTopicPartition, List<PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long>>>> kafkaUrlToThrottledRecords;
 
   public KafkaClusterBasedRecordThrottler(Map<String, EventThrottler> kafkaUrlToRecordsThrottler) {
     this.kafkaUrlToRecordsThrottler = kafkaUrlToRecordsThrottler;
     this.kafkaUrlToThrottledRecords = new VeniceConcurrentHashMap<>();
   }
 
-  public PubSubMessages<KafkaKey, KafkaMessageEnvelope, Long> poll(
+  public Map<PubSubTopicPartition, List<PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long>>> poll(
       PubSubConsumer consumer,
       String kafkaUrl,
       long pollTimeoutMs) {
-    PubSubMessages<KafkaKey, KafkaMessageEnvelope, Long> consumerRecords = kafkaUrlToThrottledRecords.get(kafkaUrl);
+    Map<PubSubTopicPartition, List<PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long>>> consumerRecords =
+        kafkaUrlToThrottledRecords.get(kafkaUrl);
     if (consumerRecords == null) {
       consumerRecords = consumer.poll(pollTimeoutMs);
       if (consumerRecords == null) {
-        consumerRecords = PubSubMessages.empty();
+        consumerRecords = new HashMap<>();
```

Comment:
Can we return `Collections.emptyMap()` here instead of allocating a new map? Let's do the same on line 74 as well, below.

Reply:
Sure, I will replace all of these maps.